src/Controller/EloquaAppCloudEndpointController.php

```diff
@@ -0,0 +1,414 @@
+   */
+  public function execute($eloquaAppCloudService) {
+    $query = $this->request->query->all();
+    $instanceId = $this->getInstanceId($query);
+    // Get the execution ID
+    $executionId = $this->request->get("executionId");
+    if (empty($executionId)) {
+      $this->logger->error('No executionId found for @eloqua_app_cloud_service.', ['@eloqua_app_cloud_service' => $eloquaAppCloudService]);
+      // It apparently does not matter what we return here, since Eloqua does not have any way of handing external errors?
+      return new HtmlResponse('No executionId found.', 500);
+    }
+    // Now load the JSON payload form Eloqua.
+    $content = $this->request->getContent();
+
+    if (empty($content)) {
+      $this->logger->error('No content found');
+
+      // It apparently does not matter what we return here, since Eloqua does not have any way of handing external errors?
+      return new HtmlResponse('Error no no content found.', 500);
+    }
+    $payload = json_decode($content);
+    $records = $payload->items;
+    $this->logger->debug('Received execute service hook with payload @records', [
+      '@record' => print_r($records, TRUE),
+    ]);
+    $pluginReferences = $this->getEntityPlugins($eloquaAppCloudService);
+
+    // Iterate over the ServiceEntity plugins and then over the payload items.
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+      // Is this a sync or async (bulk) plugin? If the annotation is empty then assume that it is async.
+      if (!empty($plugin->respond()) && $plugin->respond() === 'synchronous') {
+        $this->logger->debug('The @plugin plugin requested a synchronous response.', ['@plugin' => $plugin->getPluginId()]);
+        // Merge all the responses into one array.
+        // TODO: Will this even work?
+        $response = $this->respondSynchronously($plugin, $records, $instanceId, $executionId, $query);
+        $responseHtml = $this->renderer->renderRoot($response);
+        $result = new CacheableResponse($responseHtml);
+        $result->addCacheableDependency($response);
+        return $result;
+      }
+      else {
+        $this->logger->debug('The @plugin plugin requested an asynchronous response.', ['@plugin' => $plugin->getPluginId()]);
+        $response = $this->respondAsynchronously($plugin, $records, $instanceId, $executionId, $query);
+        $json = new JsonResponse($response);
+        $json->setStatusCode(204);
+      }
+    }
+    return $json;
+  }
+
+  /**
+   * @param $plugin
+   * @param $records
+   * @param $instanceId
+   *
+   * @return \stdClass
+   */
+  protected function respondSynchronously($plugin, $records, $instanceId, $executionId, $query) {
+    $response = new \stdClass();
```

Comment:
- [ ] Can probably delete this line since you're assigning $response just below this comment.